contracts/strategies/donation-voting/DonationVotingStrategy.sol

```diff
@@ -0,0 +1,451 @@
+            revert NOT_ALLOWED();
+        }
+
+        IAllo.Pool memory pool = allo.getPool(poolId);
+        if (pool.amount < _amount) {
+            revert UNAUTHORIZED();
+        }
+
+        // TODO: This can cause pool owner stealing funds
+
+        allo.decreasePoolTotalFunding(poolId, _amount);
+        _transferAmount(pool.token, msg.sender, _amount);
+    }
+
+    /// ====================================
+    /// ============ Internal ==============
+    /// ====================================
+
+    /// @notice Returns status of the pool
+    function _isPoolActive() internal view override returns (bool) {
+        if (registrationStartTime <= block.timestamp && block.timestamp <= registrationEndTime) {
+            return true;
+        }
+        return false;
+    }
+
+    /// @notice Submit application to pool
+    /// @param _data The data to be decoded
+    /// @param _sender The sender of the transaction
+    function _registerRecipient(bytes memory _data, address _sender)
+        internal
+        override
+        onlyActiveRegistration
+        returns (address recipientId)
+    {
+        if (_recipients[recipientId].recipientStatus != InternalRecipientStatus.Accepted) {
+            revert RECIPIENT_ALREADY_ACCEPTED();
+        }
+
+        address recipientAddress;
+        bool isRegistryIdentity;
+        Metadata memory metadata;
+
+        // decode data custom to this strategy
+        if (registryGating) {
+            (recipientId, recipientAddress, metadata) = abi.decode(_data, (address, address, Metadata));
+
+            if (!_isIdentityMember(recipientId, _sender)) {
+                revert UNAUTHORIZED();
+            }
+        } else {
+            (recipientAddress, isRegistryIdentity, metadata) = abi.decode(_data, (address, bool, Metadata));
+            recipientId = _sender;
+            if (isRegistryIdentity && !_isIdentityMember(recipientId, _sender)) {
+                revert UNAUTHORIZED();
+            }
+        }
+
+        if (recipientAddress == address(0)) {
+            revert INVALID();
+        }
+
+        Recipient storage recipient = _recipients[recipientId];
+
+        if (_recipients[recipientId].recipientAddress == address(0)) {
+            // new recipient
+            recipient.recipientAddress = recipientAddress;
+            recipient.isRegistryIdentity = registryGating ? true : isRegistryIdentity;
+            recipient.recipientStatus = InternalRecipientStatus.Pending;
+        } else {
+            // existing recipient
+            recipient.recipientAddress = recipientAddress;
+            if (recipient.recipientStatus == InternalRecipientStatus.Rejected) {
+                recipient.recipientStatus = InternalRecipientStatus.Appealed;
+                emit Appealed(recipientId, _data, _sender);
+            } else {
+                recipient.recipientStatus = InternalRecipientStatus.Pending;
+                emit Registered(recipientId, _data, _sender);
+            }
```

Comment:
can we just have one event and emit the status? 
this  would enable us to use ternary operator instead of if-else

Reply:
Registered is offered by the interface  and appeal feels custom to the strategy, How would we combine them ?
I'm trying to use the existing events wherever we can